pandas/core/internals/blocks.py

```diff
@@ -1719,6 +1719,10 @@ def _unwrap_setitem_indexer(self, indexer):
                 elif lib.is_integer(indexer[1]) and indexer[1] == 0:
                     # reached via setitem_single_block passing the whole indexer
                     indexer = indexer[0]
+
+                elif com.is_null_slice(indexer[1]):
+                    indexer = indexer[0]
```

Comment:
Small opinion. Is it better if we combine together condition like this?
```
elif (lib.is_integer(indexer[1]) and indexer[1] == 0) or (com.is_null_slice(indexer[1])):
     # reached via setitem_single_block passing the whole indexer
     indexer = indexer[0]
```

Reply:
Put it into two because this is easier to read imo